src/app/Controllers/MoviesController.cs

```diff
@@ -96,37 +85,37 @@ private string GetMethodText(string q, string genre, int year, double rating, st
                 // add the query parameters to the method name if exists
                 if (HttpContext.Request.Query.ContainsKey("q"))
                 {
-                    method = string.Format(CultureInfo.InvariantCulture, $"{method}:q:{q}");
+                    method = string.Format(CultureInfo.InvariantCulture, $"{method}:q:{movieQueryParameters.Q}");
```

Comment:
should we check the movie query parameter class instead of the request.query?

Reply:
Some of the properties can be checked for values to set the method, however others (such as `int`) will have default values. it would take more code to determine this versus what was already built.